services/QuillLMS/client/app/bundles/Teacher/components/general_components/activity_icon_with_tooltip.jsx

```diff
@@ -4,10 +4,56 @@ import React from 'react';
 import { nonRelevantActivityClassificationIds, } from '../../../../modules/activity_classifications';
 import { requestGet, } from '../../../../modules/request/index';
 import { closedLockIcon, scheduledIcon, } from '../../../Shared/index';
-import activityFromClassificationId from '../modules/activity_from_classification_id.js';
 import ScorebookTooltip from '../modules/componentGenerators/tooltip_title/scorebook_tooltip_title';
 import gradeColor from '../modules/grade_color.js';
 
+function skillDescription(grade) {
+  if (grade == null) {
+    return 'did not complete the activity';
+  } else if (grade < 0.32) {
+    return 'rarely demonstrated the activity\'s skill';
+  } else if (grade < 0.83) {
+    return 'sometimes demonstrated the activity\'s skill';
+  } else if (grade <= 1.0) {
+    return 'frequently demonstrated the activity\'s skill';
+  }
+  return 'did not complete the activity';
+};
+
+function activityFromClassificationId(classificationId) {
+  const intClassificationId = parseInt(classificationId);
+  if (intClassificationId === 1) {
+    return 'proofreader';
+  } else if (intClassificationId === 2) {
+    return 'grammar';
+  } else if (intClassificationId === 4) {
+    return 'diagnostic';
+  } else if (intClassificationId === 5) {
+    return 'connect';
+  } else if (intClassificationId === 6) {
+    return 'lessons';
+  } else if (intClassificationId === 9) {
+    return 'evidence';
+  }
+};
+
+function activityIconDescription(classificationId) {
+  const intClassificationId = parseInt(classificationId);
+  if (intClassificationId === 1) {
+    return 'Flag representing Quill Proofreader';
+  } else if (intClassificationId === 2) {
+    return 'Puzzle piece representing Quill Grammar';
+  } else if (intClassificationId === 4) {
+    return 'Magnifying glass representing Quill Diagnostic';
+  } else if (intClassificationId === 5) {
+    return 'Bullseye representing Quill Connect';
+  } else if (intClassificationId === 6) {
+    return 'Apple representing Quill Lessons ';
+  } else if (intClassificationId === 9) {
+    return 'Book to represent Quill Reading for Evidence';
+  }
+};
```

Comment:
I don't know if this is worth refactoring, but a way to approach this without all the `else if` clauses would be to make an object that's keyed by `classificationId` and return the value at that key:
```javascript
function activityFromClassificationId(classificationId)
  const intClassificationId = parseInt(classificationId);
  const classificationNames = {
    1: 'proofreader',
    2: 'grammar',
    4: 'diagnostic',
    5: 'connect',
    6: 'lessons',
    9: 'evidence'
  }
  classificationNames[intClassificationId]
}
```